metagraph/src/cli/query.cpp

```diff
@@ -313,6 +314,29 @@ slice_annotation(const AnnotatedDBG::Annotator &full_annotation,
                               utils::LessFirst(), num_threads);
     }
 
+    if (const auto *mat = dynamic_cast<const IntMatrix *>(&full_annotation.get_matrix())) {
+        std::vector<uint64_t> row_indexes;
+        row_indexes.reserve(full_to_small.size());
+        for (const auto &[in_full, _] : full_to_small) {
+            assert(in_full < full_annotation.num_objects());
+            row_indexes.push_back(in_full);
+        }
+
+        auto slice = mat->get_row_values(row_indexes);
```

Comment:
how would this perform if it was done in batches with multiple threads?

Reply:
One would need to test that. My focus was the single-thread mode.
Multithreading can be implemented inside `get_row_values` or here, as with binary annotations. One would need to do these experiments to make it parallel.